### ISSUE TITLE
Plugins for Wallet GAS estimation on contract-specific invocations

### ISSUE THREAD

**Issue by igormcoelho** [neo-project/neo#2448]
**Summary or problem description**
As pointed out by @nicolegys and @ArithMegatron, there can be some practical challenges on using `Refuel` operation (giving more GAS during execution).
In my opinion, the best way to provide precise GAS estimation for contract-specific setups, is to allow the development of Wallet plugins that are made specifically for contract-specific invocations.
For example, I develop some NEP-17, and my operations require some specific amount of GAS, or some other data treatment to *make users spend less* (contracts will work anyway, we are just discussing ways of being *more efficient*). So they insert their .dll, .so, or any other binary/code extension into wallet (or some script?), that is invoked just for their operations.

**Do you have any solution you want to propose?**
Allow Wallet extensions for contract-specific invocations (such as NEP-17), that provide precise GAS calculations.

**Neo Version**
- Neo 3

**Where in the software does this update applies to?**
- Other: Wallet

**Comment by igormcoelho** [comment]
See https://github.com/neo-project/neo/pull/2444

**Comment by roman-khimov** [comment]
> I provide special .dll, .so, or any other binary/code extension into their wallet (or some script?), that is invoked just for their operations

1. Won't work with any non-C# node/wallet.
2. I already envision "Install THIS plugin for FREE Neo N3 transactions right NOW!!!" things going on.

**Comment by igormcoelho** [comment]
I agree with both points @roman-khimov , and specially  "Install THIS plugin for FREE Neo N3 transactions right NOW!!!" hahaha 😂 

But this only happens if users install unsupported plugins/wallets, and that's why this issue is important: to discuss safe interfaces for doing that. If we fail to provide that, rogue contract-specific wallets that provide cheap or free operations will prevail.
